src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -493,6 +504,40 @@ private static bool IsValueTypeErrorOrBlank<T>(FormulaValue val)
             return val is T || val is BlankValue || val is ErrorValue;
         }
 
+        private static FormulaValue DistinctValueType(List<(DValue<RecordValue> row, FormulaValue distinctValue)> pairs, IRContext irContext)
+        {
+            var lookup = new HashSet<string>();
+            var result = new List<DValue<RecordValue>>();
+            var name = ((TableType)irContext.ResultType).SingleColumnFieldName;
+            var type = irContext.ResultType;
+
+            foreach (var (row, distinctValue) in pairs)
+            {
+                if (distinctValue is ErrorValue errorValue)
+                {
+                    return errorValue;
+                }
+
+                var key = distinctValue.ToObject();
+
+                if(key != null && key is not object)
+                {
+                    return CommonErrors.RuntimeTypeMismatch(irContext);
+                }
+
+                var keySerialized = System.Text.Json.JsonSerializer.Serialize(key);
```

Comment:
>System.Text.Json.JsonSerializer.Serialize(key); [](http://example.com/codeflow?start=36&length=47)

We should avoid the json call (see #884).
Beware that Json isn't necessarily canonical -the same value can serializer as multiple different strings. 

But we could use FormulaValue.ToExpressionValue!
That also gets a string value.

Reply:
Although ... why do we need to stringify at all? Can't we just use `HashSet<object>(); ` and object.equals?

Reply:
Since now we are not using Agreegate types, moved back to `HashSet<object>();`